core/models.py

```diff
@@ -312,3 +322,86 @@ class ComputedFeatureDrift(models.Model):
 
     def __str__(self):
         return str(self.reference_project_id)
+
+
+class NLPRequest(BaseModel):
+    class RequestStatus(models.IntegerChoices):
+        # NOTE: If changed here, corresponding values should be changed in DEEP as well.
+        INITIATED = 1
+        SUCCESS = 2
+        FAILED = 3
+        PROCESS_INPUT_URL_FAILED = 4
+
+    class FeaturesType(models.TextChoices):
+        NGRAMS = "ngrams", "Ngrams"
+        TOPICMODEL = "topicmodel", "Topicmodel"
+        SUMMARIZATION = "summarization", "Summarization"
+        GEOLOCATION = "geolocation", "Geolocation"
+        TAGS_MAPPING = "tags-mapping", "Tags Mapping"
+        PREDICTION = "prediction", "Prediction"
```

Comment:
maybe CLASSIFICATION_MODEL_PREDICTION for clarity?

Reply:
Renamed to ENTRY_CLASSIFICATION